File: gmao-backend/src/main/java/gmao_backend/measure/Measure.java
package com.gmao.gmao_backend.measure;

import com.gmao.gmao_backend.unit.MeasurementUnit;

import jakarta.persistence.*;
import lombok.*;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(
        name = "measures",
        uniqueConstraints = {
                @UniqueConstraint(
                        name = "uk_measure_code",
                        columnNames = "code"
                )
        }
)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Measure {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 255)
    private String name;

    @Column(nullable = false, unique = true, length = 100)
    private String code;

    @Column(columnDefinition = "TEXT")
    private String description;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "unit_id", nullable = false)
    private MeasurementUnit unit;

    @CreationTimestamp
    @Column(
            name = "created_at",
            nullable = false,
            updatable = false
    )
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}

File: gmao-frontend/src/types/measure.ts
import type { UnitType } from "./unit";

export interface Measure {
  id: number;
  name: string;
  code: string;
  description: string | null;

  unitId: number;
  unitName: string;
  unitSymbol: string;
  unitType: UnitType;

  createdAt: string;
  updatedAt: string;
}

export interface CreateMeasureRequest {
  name: string;
  code: string;
  description: string;
  unitId: number;
}

export interface UpdateMeasureRequest {
  name: string;
  code: string;
  description: string;
  unitId: number;
}